### ISSUE TITLE
Support for ECS

### ISSUE THREAD

**Issue by simeonradivoev** [ssannandeji/Zenject-2019#541]
Some helper functions for integration with unity's ECS would be helpful. Here are some possible cases: 

`Container.Bind<System>().FromEcs()`
`Container.Bind<System>().FromNewInEcs("World Name")`
`Container.Bind<System>().FromEcs("World Name")`
`Container.Bind<System>().FromNewInEcs()`

maybe some  helpers for creating archetypes.

`Container.BindArchetype<Class,Class2>()`

**Comment by jianmingyong** [comment]
I don't think it would work nicely right now with unity still fiddling with constant updates of their ecs code. When it is more mature, maybe it would be more feasible without getting break when they finalized.
